app/models/bulk.py

```diff
@@ -9,48 +14,123 @@ class CsvContainer(db.Model):
-        return self.csv_rows[0]
+        return '%s' % self.csv_rows[row_num].csv_body_cells[cell_num].data
 
-    def content_rows_iter(self):
-        i = iter(self.csv_rows)
-        next(i)
-        return i
+    def predict_options(self):
+        for i, column in enumerate(self.csv_header_row.csv_header_cells):
+            if column.descriptor_type == 'option':
+                predicted_options = set()
+                for j in range(len(self.csv_rows)):
+                    predicted_options.add(self.cell_data(j, i))
+                column.predicted_options = predicted_options
+                db.session.add(column)
+        db.session.commit()
 
     def __repr__(self):
         return '<CsvContainer \'%s\'>' % self.file_name
 
+    @staticmethod
+    def most_recent(user):
+        return CsvContainer.query.filter_by(user=user).order_by(
+            desc(CsvContainer.date_uploaded)
+        ).limit(1).first()
+
+
+class CsvHeaderRow(db.Model):
+    """
+    Schema for a header row in a CSV file. Cells should be added in the order
+    that they appeared in the original row of the CSV file.
+    """
+    __tablename__ = 'csv_header_rows'
+    id = db.Column(db.Integer, primary_key=True)
+    csv_container_id = db.Column(db.Integer,
+                                 db.ForeignKey('csv_containers.id'))
+    csv_header_cells = db.relationship('CsvHeaderCell',
+                                       backref='csv_header_row',
+                                       uselist=True, cascade='delete')
+
 
-class CsvRow(db.Model):
+class CsvBodyRow(db.Model):
     """
-    Schema for a row in a CSV file. Cells should be added in the order that
-    they appeared in the original row of the CSV file.
+    Schema for a content row in a CSV file. Cells should be added in the order
+    that they appeared in the original row of the CSV file.
     """
-    __tablename__ = 'csv_rows'
+    __tablename__ = 'csv_body_rows'
     id = db.Column(db.Integer, primary_key=True)
-    csv_cells = db.relationship('CsvCell', backref='csv_row', uselist=True)
     csv_container_id = db.Column(db.Integer,
                                  db.ForeignKey('csv_containers.id'))
+    csv_body_cells = db.relationship('CsvBodyCell', backref='csv_body_row',
+                                     uselist=True, cascade='delete')
+
+
+class CsvHeaderCell(db.Model):
+    """
+    Schema for a header cell in a CSV file. Each cell contains one
+    comma-separated string in the first row of a CSV file.
+    """
+    __tablename__ = 'csv_header_cells'
+    id = db.Column(db.Integer, primary_key=True)
+    csv_header_row_id = db.Column(db.Integer,
+                                  db.ForeignKey('csv_header_rows.id'))
+    data = db.Column(db.Text)
+    descriptor_type = db.Column(db.Integer)  # 'option' or 'text'
+    predicted_options = db.Column(db.PickleType)  # Set of options (strings)
+    new_options = db.Column(db.PickleType)  # Set of options (strings)
```

Comment:
Describing model contents is actually the one place where I would advocate for comments next to the line (esp. these one-liners), mostly because most models don't need comments and these are only to clarify behavior.